### ISSUE TITLE
Http tests failed because of HttpListener

### ISSUE THREAD

**Issue by wfurt** [dotnet/corefx#40632]
We got catastrophic failure without any test results posted:

https://helix.dot.net/api/2019-06-17/jobs/84ef369e-9795-4f92-bfef-43a2f1558542/workitems/System.Net.Http.Functional.Tests/console

Fortunately we got core dump collected: https://helix.dot.net/api/2019-06-17/jobs/47e2ed6c-bcd0-4f73-bed9-be21d9dda28a/workitems/System.Net.Http.Functional.Tests/files/dotnet.exe.4484.dmp

```
System.Net.Http.Functional.Tests.PlatformHandler_HttpClientHandler_Proxy_Test.Proxy_UseEnvironmentVariableToSetSystemProxy_RequestGoesThruProxy [SKIP]
      Test needs SocketsHttpHandler
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Net.HttpListener.DisconnectAsyncResult.IOCompleted(DisconnectAsyncResult asyncResult, UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped) in /_/src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs:line 1957
   at System.Net.HttpListener.DisconnectAsyncResult.WaitCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped) in /_/src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs:line 1969
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pNativeOverlapped) in /_/src/System.Private.CoreLib/src/System/Threading/Overlapped.cs:line 64
----- end Tue 08/27/2019 19:12:57.89 ----- exit code -1073741819 ----------------------------------------------------------


Searching 'C:\dotnetbuild\work\84ef369e-9795-4f92-bfef-43a2f1558542\Work\f71c1b3b-8b72-4b86-bd6d-38f6b6265b6f\Exec\..' for log files
Found log 'C:\dotnetbuild\work\84ef369e-9795-4f92-bfef-43a2f1558542\Work\f71c1b3b-8b72-4b86-bd6d-38f6b6265b6f\Exec\..\console.b09eba4f.log'
Uri 'https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-corefx-refs-pull-40543-merge-84ef369e97954f92bf/System.Net.Http.Functional.Tests/console.b09eba4f.log?sv=2018-03-28&sr=c&sig=Yow60lVW%2FgDZ9OgiHyNGfRR%2F2wWLR2X4KA0brC0whvU%3D&se=2019-09-06T18%3A52%3A28Z&sp=rl'
Generated log list: <ul><li><a href='https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-corefx-refs-pull-40543-merge-84ef369e97954f92bf/System.Net.Http.Functional.Tests/console.b09eba4f.log?sv=2018-03-28&sr=c&sig=Yow60lVW%2FgDZ9OgiHyNGfRR%2F2wWLR2X4KA0brC0whvU%3D&se=2019-09-06T18%3A52%3A28Z&sp=rl' target='_blank'>console.b09eba4f.log</a></li></ul>
Searching 'C:\dotnetbuild\work\84ef369e-9795-4f92-bfef-43a2f1558542\Work\f71c1b3b-8b72-4b86-bd6d-38f6b6265b6f\Exec' for test results files
No results file found in any of the following formats: xunit, junit, trx
 Tue 08/27/2019-19:12:58.69
2019-08-27 19:12:59,004: ERROR: xunit-reporter(69): main: Unable to report xunit results: no test results xml file found.
```

it would be great iof somebody can take a look before the link corefile link expires:

cc: @scalablecory @davidsh

**Comment by stephentoub** [comment]
This is causing a bunch of outerloop failures, e.g.
https://helix.dot.net/api/2019-06-17/jobs/9057a15a-c682-4fca-a2ec-edf1cb2f135b/workitems/System.Net.Http.Functional.Tests/files/console.038ba47f.log
```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Net.HttpListener.DisconnectAsyncResult.IOCompleted(DisconnectAsyncResult asyncResult, UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped) in /_/src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs:line 1957
   at System.Net.HttpListener.DisconnectAsyncResult.WaitCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped) in /_/src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs:line 1969
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pNativeOverlapped) in /_/src/System.Private.CoreLib/src/System/Threading/Overlapped.cs:line 64
```

**Comment by wfurt** [comment]
I can revert #40466 unless @scalablecory is getting closer on improvement.

**Comment by scalablecory** [comment]
PR #40783 to revert so we can stop test failures. I need to look at the overall design to understand what correct long-term fix is.

**Comment by scalablecory** [comment]
Change reverted; closing this issue. Tracking final fix in original issue #39552.
